apps/ui/src/pages/App.tsx

```diff
@@ -29,7 +29,8 @@ function App() {
 			<Route element={<PageLayout />}>
 				<Route index element={<HomePage />} />
 				<Route path="dashboard" element={<DashboardPage />} />
-				<Route path="application/:id" element={<ApplicationViewer />} />
+				<Route path="application/:id" element={<ApplicationViewer isEditMode={false} />} />
+				<Route path="application/:id/edit" element={<ApplicationViewer isEditMode />} />
```

Comment:
I mean either way works! I'm personally always a fan of having a route for stuff like editing (even if it is the same component with a switch flag), since it feels like a big enough action (and UI change for the user). But I guess that's just a matter of aesthetics than anything else. 

Personally for me, I when I share URLs with others, and I find it easier to strip `/edit` out rather than fiddling with query params, but that might be a me thing haha

Would having it as it's own route affect any other implementation stuff?

Reply:
Either way works!
I'm just thinking what happens in the scenario that the user links the applicationViewer as edit `applications/:id/edit` but the application state is in viewing mode? Would they be rerouted back to viewing? Would they be allowed to edit fields still? 

Either way, we would be checking from the API response if the user is in draft or not. So I guess the question is:

Do we want to protect `application/:id/edit` route and if the user links it while not in draft, send it back to view only route?
or, dynamically switch the state of view/edit based off the response object and have some sort of flag `isEdit` living in some sort of store which will tell the components what to render?

Reply:
Ah I see, you bring up a good point. 

A way we could handle this is just to check if the application is in draft mode on load of the page, since we're already grabbing info for the page, that's an easy check to do. If it is, then we continue, if not, we can reroute to view mode, **OR**, we could also display that "Do you want to edit this application" modal if we detect that discrepancy. We could then set whatever is the appropriate state that way?

Reply:
Protecting the page and having logic from the response dictate the redirect in the protection rules would work! I do have some thoughts.

I have questions about automatically displaying the "Do you want to edit this application" as it might expand the scope a little and might have to consult Patrick and Alexis for this feature. The "Edit" button will work different on the view page then if we link the a user directly to the edit page. On the applications page(view) it would function as what you defined, cancel to close modal and Edit to redirect and change state. But if the user gets linked to the `Edit` page, the cancel would no longer just close the modal and it would also redirect back to the view.
 So we would probably end up creating a new modal for this use-case. It just seems to me a little more effort into this solution.

I'm also concerned about prop drilling `isEditableMode`. We will be needing that flag in lots of different components and some will be pretty deeply nested. I think we would end up using a store anyways for this applications page.

If i'm missing something please let me know! This is a great conversation to have going forward with our application viewer!